nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/operation/HotReloadStudioSnapshot.java

```diff
@@ -71,46 +80,110 @@
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         try {
             updateInProgress = true;
-            hotReloadPackage(snapshotPkg);
+            hotReloadPackage(snapshotPkg, resultJSON);
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         } finally {
             updateInProgress = false;
         }
     }
 
-    public void hotReloadPackage(DownloadablePackage remotePkg) {
+    public void hotReloadPackage(DownloadablePackage remotePkg, JSONObject resultJSON) {
+
         if (validate) {
             pm.flushCache();
 
             String targetPlatform = PlatformVersionHelper.getPlatformFilter();
             if (!TargetPlatformFilterHelper.isCompatibleWithTargetPlatform(remotePkg, targetPlatform)) {
-                throw new NuxeoException(
+                resultJSON.put("status", "error");
+                resultJSON.put("message",
                         String.format("This package is not validated for your current platform: %s", targetPlatform));
+                return;
+            }
+
+            PackageDependency[] pkgDeps = remotePkg.getDependencies();
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("%s target platforms: %s", remotePkg,
+                        ArrayUtils.toString(remotePkg.getTargetPlatforms())));
+                log.debug(String.format("%s dependencies: %s", remotePkg, ArrayUtils.toString(pkgDeps)));
+            }
+
+            String packageId = remotePkg.getId();
+
+            // check deps requirements
+            if (pkgDeps != null && pkgDeps.length > 0) {
+                DependencyResolution resolution = pm.resolveDependencies(packageId, targetPlatform);
+                if (resolution.isFailed() && targetPlatform != null) {
+                    // retry without PF filter in case it gives more information
+                    resolution = pm.resolveDependencies(packageId, null);
+                }
+                if (resolution.isFailed()) {
+                    resultJSON.put("status", "dependency_error");
+                    resultJSON.put("message",
+                            String.format("Dependency check has failed for package '%s' (%s)", packageId, resolution));
+                    return;
+                } else {
+                    List<String> pkgToInstall = resolution.getInstallPackageIds();
+                    if (pkgToInstall != null && pkgToInstall.size() == 1 && packageId.equals(pkgToInstall.get(0))) {
+                        // ignore
+                    } else if (resolution.requireChanges()) {
+                        // do not install needed deps: they may not be hot-reloadable and that's not what the
+                        // "update snapshot" button is for.
+                        // status.addError(resolution.toString().trim().replaceAll("\n", "<br />"));
```

Comment:
To delete ?

Reply:
should be kept, i think, missed it

Reply:
indeed

Reply:
actually discussed live, not needed in this operation, can be removed // maybe still explained in a comment